node/util/manifest.ts

```diff
@@ -155,3 +114,31 @@ export async function makeDefaultManifest(
 
   return manifest
 }
+
+export function updateManifest(
+  manifest: Manifest,
+  uploadFile: UploadFile,
+  version: string
+) {
+  manifest.version = version
+  manifest.dependencies = updateDependencies(manifest.dependencies, uploadFile)
+
+  return manifest
+}
+
+function updateDependencies(
+  dependencies: { [key: string]: string },
+  uploadFile: UploadFile
+) {
+  // tslint:disable-next-line:forin
+  for (const element of uploadFile.dependencies) {
+    const [depAndVar] = element.split(':')
+    const [dep] = depAndVar.split('@')
+    // eslint-disable-next-line prefer-destructuring
+    const ver = depAndVar.split('@')[1]
```

Comment:
I think you can destruct the array in two variables:

```suggestion
    const [dependency, version] = depAndVar.split('@')
```

Note: I think we need to check that version exists next line.